src/perform_step/sdirk_perform_step.jl

```diff
@@ -367,23 +373,20 @@ end
 
   # TODO: Add extrapolation
   @.. zᵧ = zprev
-  nlsolver.z = zᵧ
-
+  z .= zᵧ
   @.. tmp = uprev + d*zprev
   nlsolver.c = γ
-  zᵧ = nlsolve!(integrator, cache)
+  zᵧ .= nlsolve!(integrator, cache)
```

Comment:
interesting. Was there an accidental harmless aliasing here before?

Reply:
I think there were some pointer swapping things before which were dependent on the fact that we store two references to the same array therefore they worked. Now we have single reference, I would have to see if I can re-do those tricks